timer/src/main/java/com/toggl/timer/startedit/domain/StartEditReducer.kt

```diff
@@ -462,11 +466,13 @@ class StartEditReducer @Inject constructor(
 
     private fun EditableTimeEntry.shouldStart() = this.ids.isEmpty()
     private fun EditableTimeEntry.endTime(now: OffsetDateTime): OffsetDateTime {
-        if (startTime == null)
-            return now
-
-        val relativeDuration = duration ?: startTime.absoluteDurationBetween(now)
-        return startTime + relativeDuration
+        return startTime.let { startTime ->
+            if (startTime == null) now
+            else {
+                val relativeDuration = duration ?: startTime.absoluteDurationBetween(now)
+                startTime + relativeDuration
+            }
+        }
```

Comment:
```suggestion
        return startTime?.let { startTime ->
                val relativeDuration = duration ?: startTime.absoluteDurationBetween(now)
                startTime + relativeDuration
            } ?: now
        }
```

Another opportunity 🤔 What is our standing on the elvis operator again? 🤔 Maybe we should discuss when we should use it 🤔

Reply:
I'm not against, especially when it helps reduce nesting.